test/langtools/jdk/javadoc/doclet/testSnippetTag/SnippetTester.java

```diff
@@ -87,17 +90,10 @@ protected static void addSnippetFile(Path srcDir, String packageName, String fil
         }
     }
 
-    // TODO: perhaps this method could be added to JavadocTester
     protected void checkOutputEither(Output out, String first, String... other) {
-        checking("checkOutputEither");
-        String output = getOutput(out);
-        Stream<String> strings = Stream.concat(Stream.of(first), Stream.of(other));
-        Optional<String> any = strings.filter(output::contains).findAny();
-        if (any.isPresent()) {
-            passed(": following text is found:\n" + any.get());
-        } else {
-            failed(": nothing found");
-        }
+        var strings = Stream.concat(Stream.of(first), Stream.of(other))
+                .toArray(String[]::new);
+        new OutputChecker(out).checkAnyOf(strings);
```

Comment:
This works, and is a clever solution for now.  Maybe "one of these days" we remove the `checkOutputEither` method.